app/src/main/java/com/kmarlow/custominstrumentation/sdk/AndromiumControllerService.java

```diff
@@ -0,0 +1,198 @@
+        super(controllerService, launchIntent, appId);
+        this.controllerService = controllerService;
+        this.stackManager = new ActivityStackManager();
+
+        // Uncomment to check for leaks
+        // RefWatcher refWatcher = InstApplication.getRefWatcher(controllerService);
+        // refWatcher.watch(this);
+
+        Pair<AndromiumInstrumentation, ActivityLifecycleManager> pair = AndromiumInstrumentationInjector.inject(controllerService, this);
+        if (pair == null) {
+            throw new IllegalStateException("Unable to inject Andromium instrumentation");
+        }
+        this.instrumentation = pair.first;
+        this.lifecycleManager = pair.second;
+
+        controllerService.initWindow(this, appId);
+
+        // TODO: Call into the LAUNCHER activity
+        // startLauncherActivity();
+        Intent intent = new Intent(controllerService, SubActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        controllerService.startActivity(intent);
+    }
+
+    private void startLauncherActivity() {
+        final PackageManager pm = controllerService.getPackageManager();
+        Intent launcherIntent = pm.getLaunchIntentForPackage(context.getPackageName());
+
+        if (launcherIntent == null) {
+            throw new IllegalStateException("No launcher activity found for " + context.getPackageName());
+        }
+
+        ComponentName componentName = launcherIntent.getComponent();
+        // Make sure to clear the stack of activities.
+        Intent mainIntent = IntentCompat.makeRestartActivityTask(componentName);
+        context.startActivity(mainIntent);
+    }
+
+    @Override
+    public int getAppBodyLayoutXml() {
+        // No-op
+        return 0;
+    }
+
+    @Override
+    public void initializeAndPopulateBody(View view) {
+        // No-op
+    }
+
+    @Override
+    public boolean onKeyEvent(KeyEvent event) {
+        ActivityRecord current = stackManager.peekTop();
+        if (current != null && current.activity != null) {
+            Activity currentActivity = current.activity;
+            return currentActivity.dispatchKeyEvent(event);
+        }
+
+        return super.onKeyEvent(event);
+    }
+
+    @Override
+    public WindowConfig getWindowConfiguration() {
+        return new WindowConfig(600, 500, true);
+    }
+
+    @Override
+    public void execStartActivity(Context who, IBinder token, Intent intent) {
+        Toast.makeText(who, "Start " + intent.getComponent().getShortClassName(), Toast.LENGTH_SHORT).show();
+
+        if (controllerService.isTransitioning()) {
+            return;
+        }
+
+        String className = intent.getComponent().getClassName();
+        boolean showingScreen = stackManager.isShowingScreen(className);
+        if (showingScreen) {
+
+            ActivityRecord activityRecord = stackManager.get(className);
+            // Activity exists in back stack, just redeliver intent.
```

Comment:
doesn't this depend on the intent flags? are we going to bother with intent flags and start task, clear top and all those things yet?